File: canvas/fizz-11/lib/exportData.js
const fs = require('fs')
const converter = require('json-2-csv')

const ToCsv = (data, name) => {
  const json2csvCallback = (err, csv) => {
    if (err) throw err

    fs.writeFile(`${name}.csv`, csv, 'utf8', fsErr =>  {
      if (fsErr)
        console.log('Some error occured - file either not saved or corrupted file saved.')
      else
        console.log(`The file has been generated => ${name}.csv`)
    })
  }
  converter.json2csv(data, json2csvCallback);

}

exports.ToCsv = ToCsv

File: src/canvas/data-extractor-py/src/lib/exportData.py
import pandas as pd
import os

def ToCsv(data, name):
    df = pd.DataFrame(data)
    file_name = f"{name}.csv"
    output_path = os.getenv("CSV_OUTPUT_PATH")

    df.to_csv(f"{output_path}/{file_name}", index=False)
    print(f"The file has been generated => {file_name}")
